docs/admin/repo/perforce.mdx

```diff
@@ -150,27 +150,69 @@ To enable file-level permissions:
+#### Handling IP-based rules
+
+Perforce's `protects` table allows administrators to define fine-grained access controls based on user identities and host IP addresses. By default, Sourcegraph applies all rules from the `protects` table without considering host-specific restrictions, effectively treating all host rules as the wildcard `*`. This behavior can lead to users having unintended access to repositories or files that should be restricted based on their IP addresses.
+
+If your Perforce environment relies heavily on host-based permissions, it's crucial to configure Sourcegraph appropriately to respect these restrictions. This documentation provides detailed instructions on how to enforce or ignore host rules in Sourcegraph when integrating with Perforce.
+
+##### Default Behavior
+
+By default, Sourcegraph:
+
+- **Applies all rules** in the Perforce `protects` table.
+- **Ignores host-specific restrictions**, treating all host fields as `*`.
+
+**Implication**: Users may gain access to resources that should be restricted based on their IP addresses.
+
+##### Configuration Options
+
+To ensure Sourcegraph handles host rules according to your requirements, you have two additional options:
+
+1. **Enforce Host Rules**: Configure Sourcegraph to respect and enforce IP-based restrictions defined in the `protects` table.
+2. **Ignore Host-Specific Rules**: Configure Sourcegraph to disregard any rules with a host value other than `*`.
+
+##### Enforcing host rules
+
+If you want Sourcegraph to enforce host-specific permissions, you need to enable IP restriction enforcement in your [site configuration](https://sourcegraph.com/docs/admin/config/site_config):
 
 ```json
-    {
-      "authorization": {
-        "subRepoPermissions": true,
-        "ignoreRulesWithHost": true
-      }
+{
+  "experimentalFeatures": {
+    "subRepoPermissions": {
+      "enabled": true,
+      "enforceIPRestrictions": true
     }
+  }
+}
+```
+
+When `enforceIPRestrictions` is set to `true`, Sourcegraph will use the user's IP address to apply Perforce permissions at the user level. It uses the final `X-Forwarded-For` header in the request to identify the user's IP. Note that this header can be easily spoofed, so ensure your load balancer or proxy handles `X-Forwarded-For` headers securely.
+
+##### Ignore rules with host
+
+To ignore rules that have a host value other than `*`, set `ignoreRulesWithHost` to `true` in your **code host configuration**:
+
+```json
+{
+  "authorization": {
+    "subRepoPermissions": true,
+    "ignoreRulesWithHost": true
```

Comment:
I am a little confused on exactly what `ignoreRulesWithHost` does that is different than just having `enforceIPRestrictions`. Does this rule mean we don't just treat it as a * but we ignore the whole rule?

What is the default behavior if you don't set `ignoreRulesWithHost`? 
Should we call out that these settings are mutually exclusive?

Reply:
So here I describe the default behaviour: https://github.com/sourcegraph/docs/pull/691/files#diff-265332799615f15ca1718cf47161c2845000523dfdea9385bc4363dddc0dfc0cR163-R170

We can mention that they're mutually exclusive. The site admin config should complain about it as well since it's encoded as mutually exclusive in the schema